File: csphn_denguon4_cau3.cpp
#include<iostream>
#include<algorithms>
using namespace std;
int main() {
    int squares = 0, h, v, cnt = 0;
    cin >> n >> m;
    int x[n], y[m], l[(long long) n * (n + 1)/2], r[(long long) m *(m + 1)/2];
    for(int i = 0; i < n - 1; i++) { 
        for(int j = i + 1; i < n; i++) {
            l[++cnt] = abs(x[i] - x[j]);
        }
    }
    for(int i = 0; i <  m - 1; i++) { 
        for(int j = i + 1; i < m; i++) {
            r[++cnt] = abs(y[i] - y[j]);
        }
    }
    sort(r + 1, r + m *(m + 1)/2 + 1);
    for(int i = 0; i < m *(m + 1)/2; i++) {
        tmp = lower_bound()
    }
}

File: csphn_denguon4_cau3.py
def count_squares(x_coords, y_coords, n, m):

    squares = 0
    cnt = 0
    l = []
    r = []
    for i in range(0, n - 1):
        for j in range(i + 1, n):
            cnt += 1
            l.append(abs(x_coords[i] - x_coords[j]))
    cnt = 0
    for i in range(0, m - 1):
        for j in range(i + 1, m):
            cnt += 1
            r.append(abs(y_coords[i] - y_coords[j]))
    from bisect import bisect_left, bisect_right
    r.sort()
    for i in l:
        tmp = bisect_left(r, i)
        if tmp != n and tmp != -1:
            squares += bisect_right(r, i) - tmp
    return squares

h, v = tuple(int(i) for i in input().split())
x_coords = [int(i) for i in input().split()]
y_coords = [int(i) for i in input().split()]
print(count_squares(x_coords, y_coords, h, v))
